integration/update_field_test.go

```diff
@@ -0,0 +1,929 @@
+		},
+		"DoubleIncrementIntField": {
+			filter: bson.D{{"_id", "int32"}},
+			update: bson.D{{"$inc", bson.D{{"value", float64(1.13)}}}},
+			result: bson.D{{"_id", "int32"}, {"value", float64(43.13)}},
+		},
+		"DoubleIncrementLongField": {
+			filter: bson.D{{"_id", "int64"}},
+			update: bson.D{{"$inc", bson.D{{"value", float64(1.13)}}}},
+			result: bson.D{{"_id", "int64"}, {"value", float64(43.13)}},
+		},
+		"DoubleIntIncrement": {
+			filter: bson.D{{"_id", "double"}},
+			update: bson.D{{"$inc", bson.D{{"value", int32(1)}}}},
+			result: bson.D{{"_id", "double"}, {"value", float64(43.13)}},
+		},
+		"DoubleLongIncrement": {
+			filter: bson.D{{"_id", "double"}},
+			update: bson.D{{"$inc", bson.D{{"value", int64(1)}}}},
+			result: bson.D{{"_id", "double"}, {"value", float64(43.13)}},
+		},
+		"IntIncrement": {
+			filter: bson.D{{"_id", "int32"}},
+			update: bson.D{{"$inc", bson.D{{"value", int32(1)}}}},
+			result: bson.D{{"_id", "int32"}, {"value", int32(43)}},
+		},
+		"IntNegativeIncrement": {
+			filter: bson.D{{"_id", "int32"}},
+			update: bson.D{{"$inc", bson.D{{"value", int32(-1)}}}},
+			result: bson.D{{"_id", "int32"}, {"value", int32(41)}},
+		},
+		"IntIncrementDoubleField": {
+			filter: bson.D{{"_id", "double"}},
+			update: bson.D{{"$inc", bson.D{{"value", int32(1)}}}},
+			result: bson.D{{"_id", "double"}, {"value", float64(43.13)}},
+		},
+		"IntIncrementLongField": {
+			filter: bson.D{{"_id", "int64"}},
+			update: bson.D{{"$inc", bson.D{{"value", int32(1)}}}},
+			result: bson.D{{"_id", "int64"}, {"value", int64(43)}},
+		},
+		"LongIncrement": {
+			filter: bson.D{{"_id", "int64"}},
+			update: bson.D{{"$inc", bson.D{{"value", int64(1)}}}},
+			result: bson.D{{"_id", "int64"}, {"value", int64(43)}},
+		},
+		"LongNegativeIncrement": {
+			filter: bson.D{{"_id", "int64"}},
+			update: bson.D{{"$inc", bson.D{{"value", int64(-1)}}}},
+			result: bson.D{{"_id", "int64"}, {"value", int64(41)}},
+		},
+		"LongIncrementDoubleField": {
+			filter: bson.D{{"_id", "double"}},
+			update: bson.D{{"$inc", bson.D{{"value", int64(1)}}}},
+			result: bson.D{{"_id", "double"}, {"value", float64(43.13)}},
+		},
+		"LongIncrementIntField": {
+			filter: bson.D{{"_id", "int32"}},
+			update: bson.D{{"$inc", bson.D{{"value", int64(1)}}}},
+			result: bson.D{{"_id", "int32"}, {"value", int64(43)}},
+		},
+
+		"FieldNotExist": {
+			filter: bson.D{{"_id", "int32"}},
+			update: bson.D{{"$inc", bson.D{{"foo", int32(1)}}}},
+			result: bson.D{{"_id", "int32"}, {"value", int32(42)}, {"foo", int32(1)}},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			ctx, collection := setup(t, shareddata.Composites)
+
+			_, err := collection.InsertMany(ctx, []any{
+				bson.D{{"_id", "double"}, {"value", 42.13}},
+				bson.D{{"_id", "double-nan"}, {"value", math.NaN()}},
+				bson.D{{"_id", "int32"}, {"value", int32(42)}},
+				bson.D{{"_id", "int64"}, {"value", int64(42)}},
+			})
```

Comment:
We have value that are good enough in shareddata.Scalars, we should use them

Reply:
> We have value that are good enough in shareddata.Scalars, we should use them

done.

Reply:
May you please slow down and update other place too?